### ISSUE TITLE
pureExternalModules when using API - 0.50.0 regression

### ISSUE THREAD

**Issue by Conduitry** [rollup/rollup#1645]
This is a peculiar one - Here's as minimal a reproduction as I've found, but I've only gotten this to happen when using Rollup's programmatic API, and not when using `rollup.config.js`

`input.js`:
```javascript
import { external } from 'external'
external()
```

`test.js`:
```javascript
let { rollup } = require('rollup')

async function main() {
	console.log((await (await rollup({
		input: './input.js',
		external: ['external'],
		pureExternalModules: true,
	})).generate({ format: 'cjs' })).code)
}

main()
```

[Gist](https://gist.github.com/Conduitry/21a69fb638b9e73f69ccf9d6a12afeeb)

Running `test.js` produces
```javascript
'use strict';

external.external();
```

i.e., no `require('external')`.

If I try to generate with `format: 'es'`, I instead get this promise rejection:

```
(node:648) TypeError: replacement content must be a string
    at MagicString$1.overwrite (.../node_modules/rollup/dist/rollup.js:1164:46)
    at Node.render (.../node_modules/rollup/dist/rollup.js:7063:10)
    at eachChild.child (.../node_modules/rollup/dist/rollup.js:6001:34)
    at keys.forEach.key (.../node_modules/rollup/dist/rollup.js:5847:5)
    at Array.forEach (<anonymous>)
    at Node.eachChild (.../node_modules/rollup/dist/rollup.js:5840:13)
    at Node.render (.../node_modules/rollup/dist/rollup.js:6001:8)
    at eachChild.child (.../node_modules/rollup/dist/rollup.js:6001:34)
    at keys.forEach.key (.../node_modules/rollup/dist/rollup.js:5847:5)
    at Array.forEach (<anonymous>)
```

**Comment by gaearon** [comment]
I also bumped into this when we tried to update React to use 0.50.
Here is a repro: https://gist.github.com/gaearon/640bcdbe94f55527ced9837ce4916a13

It's almost the same as the above repro (I based on it), but we saw this with default import.
This code:

```js
import warning from 'warning';

warning('hi');
```

becomes this:

```js
'use strict';

warning('hi');
```

but should have been:

```js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var warning = _interopDefault(require('warning'));

warning('hi');
```

(which it is, without the `pureExternalModules` option)

**Comment by lukastaegert** [comment]
Hi @Conduitry, @gaearon,
sorry for letting you wait so long. It turned out that it was probably just a missed refactoring. Can you confirm that it works for you when you install from
```
github.com/lukastaegert/rollup#do-not-exclude-pure-external-modules
```
?

Side-note: Direct installation from GitHub should work fine with npm@5 but I did not get it to work with yarn i.e. yarn does not seem to run the `prepare` script necessary to create the rollup binary. If anyone has an idea what needs to be changed to make this work, any suggestion would be welcome.

**Comment by Conduitry** [comment]
@lukastaegert Yep, that's working for me, thanks!

(On a side note, I'm going to have to check out that `prepare` script thing. Installing from your branch does work for me with `npm@5` but I'm not sure how :) Does npm magically install Rollup and all its dev dependencies to some temporary location first and then copy it and the dependencies in?)

**Comment by lukastaegert** [comment]
Fixed in rollup@0.50.1
